addons/binding/org.openhab.binding.meterreader/src/main/java-extern/org/openmuc/jrxtx/JRxTxPort.java

```diff
@@ -0,0 +1,323 @@
+package org.openmuc.jrxtx;
```

Comment:
Is this a facade because jsml requires it? I would expect jsml to only depend on `gnu.io` directly?
If you implemented that facade, please also put the EPL license header at the top. If not, please consider to rewrite it, so that we can add this code as EPL here, just like the rest.

Reply:
j62056 lib depends on jrxtx lib (from openmuc) which is a facade for `gnu.io`.
Unfortunately it directly creates a `gnu.io.RXTXPort`, but for my case I need a `gnu.io.TelnetSerialPort`. For that I created a factory which can be used to decide which implementation to use.
-> Workaround now is to shadow the `org.openmuc.jrxtx.JRxTxPort` class and make use of my factories.

Reply:
> j62056 lib depends on jrxtx lib (from openmuc) which is a facade for gnu.io

So does it really depend on jrxtx or should it work with any RXTX packaging (such as nrjavaserial)?

> Unfortunately it directly creates a gnu.io.RXTXPort

Not rather a `org.openmuc.jrxtx.JRxTxPort` and that's why you shadow it?

Ok, but in any case it would be good to add a license header and a short explanation on the reason why this class is here.

Reply:
Actually rxtx library includes `gnu.io.*` classes and j62056.jar is depending on that. As we are using nrjavaserial as an implementation of `gnu.io`, we can't go that way!
-> As a workaround I included the `org.openmuc.jrxtx.JRxTxPort` which is used by j62056.jar and modified it to work with any implementation of `gnu.io.SerialPort` (formerly it was just working with the implementation `gnu.io.RXTXPort`.

Anyhow - That needs to be explained as you rightly noted